src/dbnode/persist/fs/cross_block_iterator.go

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) 2020 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package fs
+
+import (
+	"bytes"
+
+	"github.com/m3db/m3/src/dbnode/encoding"
+	"github.com/m3db/m3/src/dbnode/ts"
+	xtime "github.com/m3db/m3/src/x/time"
+)
+
+type crossBlockIterator struct {
+	idx        int
+	exhausted  bool
+	current    encoding.ReaderIterator
+	byteReader *bytes.Reader
+	records    []BlockRecord
+}
+
+// NewCrossBlockIterator creates a new CrossBlockIterator.
+func NewCrossBlockIterator(pool encoding.ReaderIteratorPool) CrossBlockIterator {
+	return &crossBlockIterator{
+		idx:        -1,
+		current:    pool.Get(),
+		byteReader: bytes.NewReader(nil),
+	}
```

Comment:
nit: I would do the following:
```go
c := &crossBlockIterator{current: pool.Get(), byteReader: bytes.NewReader(nil)}
c.Reset(nil)
return c
```

Therefore not duplicating the -1 or any other reset logic that gets added to this iterator.

Reply:
👍